tools/run-server.py

```diff
@@ -13,33 +12,44 @@
 BLUE = "\033[94m"
 
 
-def run_master():
-    with subprocess.Popen(
-        ["../master/build/determined-master", "--config-file", "master.yaml"],
-        stdout=subprocess.PIPE,
-        stderr=subprocess.STDOUT,
-    ) as proc:
-        for line in proc.stdout:
-            print(
-                f"{MAGENTA}determined-master  |{CLEAR} {line.decode('utf8')}",
-                end="",
-                flush=True,
-            )
-
-
-def wait_for_master():
+def wait_for_server(port, host='localhost', timeout=5.0):
     for _ in range(100):
```

Comment:
thought: think the issue with this current timeout is that we have to pull the db and hasura containers

I actually think we'll have a better setup for CI if we have the calls be separate for starting the cluster and the db. Because the `docker-compose up -d` for the start logs blocks while we pull the containers. One option is to keep the status quo, another one is to make separate calls to pull before calling this script

Reply:
another idea is to make the `start_db()` a blocking `subprocess.run` and then tail the logs after